pyqg/model.py

```diff
@@ -213,9 +235,11 @@ def __init__(
 
         # TODO: be more clear about what attributes are cython and what
         # attributes are python
-        PseudoSpectralKernel.__init__(self, nz, ny, nx, ntd,
-                has_q_param=int(q_parameterization is not None),
-                has_uv_param=int(uv_parameterization is not None))
+        self.kernel = kernel(nz, ny, nx,
+                q_parameterization,
+                uv_parameterization,
+                ntd,
```

Comment:
One issue -- `ntd` (number of threads) is not going to be an attribute that every kernel needs (or accepts). We could get around this by passing an optional `kernel_kwargs` argument but that seems questionable.

Normally, with dependency injection, we pass in a fully initialized object, but here we're passing in a class. That also seems questionable. Hmm.

Reply:
How do you imagine the api will work once we have multiple kernels. How will the user specify which kernel they want to use at runtime?

I would prefer to be able to write

```
model = QGModel(kernel="jax", kernel_kwargs={'foo': 'special jax argument'})
```

Reply:
Yeah, I was thinking something like that -- it would support both
```python
model = pyqg.QGModel(kernel="jax", kernel_kwargs={"foo": "special jax argument"})
```
or
```python
model = pyqg.QGModel(kernel=pyqg.JaxKernel, kernel_kwargs={"foo": "special jax argument"})
```

Reply:
However, I do think this syntax is a bit weird, since it's more normal to do dependency injection with initialized objects:

```python
model = pyqg.QGModel(kernel=pyqg.CythonFFTWKernel(fftw_num_threads=2)
model = pyqg.QGModel(kernel=pyqg.JaxKernel(foo='bar'))
```

The problem with this is that we'd need to start passing `nx`/`ny`/`nz`/other arguments to the kernel rather than the model, which is could lead to redundancies and definitely isn't backwards compatible.